### DISCUSSION TITLE
Tooltip for UI inputs

### DISCUSSION THREAD

**Discussion by helgakovacs** [marimo-team/marimo#2059] [Ideas]
**Feature request: Tooltip for UI inputs**

It would be great to have optional tooltip for the different UI inputs (for example with an information hover over icon next to the label), so the user could get more information about the input field. Having long labels to properly explain the inputs makes the app less readable.

**Comment by mscolnick** [comment]
Hi, this is a great request. I added some basic support (https://github.com/marimo-team/marimo/pull/2086), would this suffice?

**Reply by helgakovacs** [reply]
Yes, thank you for the implementation.

**Comment by daviddwlee84** [comment]
Seems we can use the trick of `<div data-tooltip="This is a tooltip"></div>` to add a tooltip in the label parameter.

Just a suggestion, I was from Streamlit, and they basically provide a "help" parameter for each input-related UI element, which I think is more intuitive to use.

---

https://docs.marimo.io/api/markdown/?h=tooltips#tooltips

You can render a tooltip by adding the data-tooltip attribute to an element.

```py
mo.md(
    '''
    <div data-tooltip="This is a tooltip">Hover over me</div>
    '''
)
mo.ui.button(
    label='<div data-tooltip="This is a tooltip">Hover over me</div>'
)
```

---

https://docs.streamlit.io/develop/api-reference/widgets/st.checkbox

> help (str or None) parameter
> A tooltip that gets displayed next to the widget label. Streamlit only displays the tooltip when label_visibility="visible". If this is None (default), no tooltip is displayed.The tooltip can optionally contain GitHub-flavored Markdown, including the Markdown directives described in the body parameter of st.markdown.
